paf-mvp-client-express/src/client-node.ts

```diff
@@ -67,417 +56,420 @@ export interface ClientNodeConfig extends Config {
+    this.app.expressApp.post(
       jsonProxyEndpoints.signPrefs,
-      cors(corsOptions),
-      checkOrigin(jsonProxyEndpoints.signPrefs),
-      (req, res) => {
-        logger.Info(jsonProxyEndpoints.signPrefs);
-        try {
-          const { identifiers, unsignedPreferences } = getPayload<PostSignPreferencesRequest>(req);
-          res.json(client.buildPreferences(identifiers, unsignedPreferences.data));
-        } catch (e) {
-          logger.Error(jsonProxyEndpoints.signPrefs, e);
-          // FIXME finer error return
-          const error: ClientNodeError = {
-            type: ClientNodeErrorType.UNKNOWN_ERROR,
-            details: '',
-          };
-          res.status(400);
-          res.json(error);
-        }
-      }
+      this.originValidator.cors,
+      this.originValidator.checkOrigin(jsonProxyEndpoints.signPrefs),
+      this.startSpan(jsonProxyEndpoints.signPrefs),
+      this.signPreferences.bind(this),
+      this.handleErrors(jsonProxyEndpoints.signPrefs),
+      this.endSpan(jsonProxyEndpoints.signPrefs)
     );
 
     // *****************************************************************************************************************
     // ***************************************************************************************************** JSON - SEED
     // *****************************************************************************************************************
-
-    app.expressApp.post(
+    this.app.expressApp.post(
       jsonProxyEndpoints.createSeed,
-      cors(corsOptions),
-      checkOrigin(jsonProxyEndpoints.createSeed),
-      (req, res) => {
-        logger.Info(jsonProxyEndpoints.createSeed);
-        try {
-          const request = JSON.parse(req.body as string) as PostSeedRequest;
-          const seed = client.buildSeed(request.transaction_ids, request.data);
-          const response = seed as PostSeedResponse; // For now, the response is only a Seed.
-          res.json(response);
-        } catch (e) {
-          logger.Error(jsonProxyEndpoints.createSeed, e);
-          // FIXME finer error return
-          const error: ClientNodeError = {
-            type: ClientNodeErrorType.UNKNOWN_ERROR,
-            details: '',
-          };
-          res.status(400);
-          res.json(error);
-        }
-      }
+      this.originValidator.cors,
+      this.originValidator.checkOrigin(jsonProxyEndpoints.createSeed),
+      this.startSpan(jsonProxyEndpoints.createSeed),
+      this.createSeed.bind(this),
+      this.handleErrors(jsonProxyEndpoints.createSeed),
+      this.endSpan(jsonProxyEndpoints.createSeed)
     );
   }
 
+  restRead(req: Request, res: Response, next: NextFunction) {
+    try {
+      const url = this.client.getReadRestUrl(req);
+      res.send(url.toString());
+      next();
+    } catch (e) {
+      const error: ClientNodeError = {
+        type: ClientNodeErrorType.UNKNOWN_ERROR,
+        details: '',
+      };
+      res.status(400);
```

Comment:
The error management will be reworked in detail soon: there would not be a big `try/catch` anymore (the exceptions would automatically be translated into `50x` by Express), and the response will be much more adequate.

However, already today all excdeptions have to be due to invalid input, so the fallback to `400` seems legit.

Reply:
Ok to postpone.

However, we don't validate the HTTP request for those endpoints before executing the business logic. So a `40x` can be inaccurate and misleading. It is even more true that there are not a lot of tests on this part of this code. Bugs happen.

Given this context, putting a `500` instead of a `40x` is the most humble approach.

Reply:
> we don't validate the HTTP request for those endpoints before executing the business logic

What do you mean? Most of the log is to validate the HTTP request (check signature).

Reply:
Ok I understand.
But if we are questioning `40x` versus `50x` then I think `40x` is more relevant.
And regarding the `x`, this will be defined in a PR focusing on error management

Reply:
We are simply saying the same thing.
It's just not the point of this PR.

Reply:
We will discuss offline. It will be simpler.
We are not saying the same thing.